api/opencl_runtime_layer.txt

```diff
@@ -604,48 +583,41 @@ sub-buffer objects created with the same buffer object is undefined.
 Only reading from both a buffer object and its sub-buffer objects or reading
 from multiple overlapping sub-buffer objects is defined.
 ====
+--
+
+[open,refpage='cl_buffer_region',desc='',type='structs',xrefs='clCreateSubBuffer']
+--
+The *cl_buffer_region* structure specifies a region of a buffer object:
+
+include::{generated}/api/structs/cl_buffer_region.txt[]
+
+  * _origin_ is the offset in bytes of the region.
+  * _size_ is the size in bytes of the region.
+
+Constraints on the values of _origin_ and _size_ are specified for the
+*<<clCreateSubBuffer>>* function to which this structure is passed.
+--
 
 
 === Reading, Writing and Copying Buffer Objects
 
 The following functions enqueue commands to read from a buffer object to
 host memory or write to a buffer object from host memory.
 
+[open,refpage='clEnqueueReadBuffer',desc='',type='protos']
+--
+To read from a buffer object to host memory, call one of the functions
 
-[source,c]
-----
-cl_int clEnqueueReadBuffer(cl_command_queue command_queue,
-                           cl_mem buffer,
-                           cl_bool blocking_read,
-                           size_t offset,
-                           size_t size,
-                           void *ptr,
-                           cl_uint num_events_in_wait_list,
-                           const cl_event *event_wait_list,
-                           cl_event *event)
-----
-
-[source,c]
-----
-cl_int clEnqueueWriteBuffer(cl_command_queue command_queue,
-                            cl_mem buffer,
-                            cl_bool blocking_write,
-                            size_t offset,
-                            size_t size,
-                            const void *ptr,
-                            cl_uint num_events_in_wait_list,
-                            const cl_event *event_wait_list,
-                            cl_event *event)
-----
-
-_command_queue_ is a valid host command-queue in which the read / write
-command will be queued.
-_command_queue_ and _buffer_ must be created with the same OpenCL context.
+include::{generated}/api/protos/clEnqueueReadBuffer.txt[]
 
-_buffer_ refers to a valid buffer object.
+include::{generated}/api/protos/clEnqueueWriteBuffer.txt[]
```

Comment:
Should there be something like

```
 [open,refpage='clEnqueueWriteBuffer',desc='',type='protos']
```

here?  I'm not sure.  My build didn't seem to create a separate `clEnqueueWriteBuffer` man page, but put that function on the `clEnqueueReadBuffer` page.  I believe this is on purpose, but `man/apispec.txt` both to be present?

This comment applies to all three read/write pairs.

Reply:
Without restructuring the spec text, I can't create two separate refpages for these pairs. I'm OK with doing that but it is out of scope for this MR. I did add alias= directives that will be used to help generate the index of refpages, though.